### PR TITLE
Simple POST endpoint

### LINKED ISSUE #9: Simple/basic server with a POST endpoint

### PR DESCRIPTION
**Pull request by nelup20**
Closes #9

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,35 +1,25 @@
...
 use std::fs;
 use std::io::prelude::*;
 use std::net::TcpListener;
 use std::net::TcpStream;
+use std::str::from_utf8;
+use request_methods::RequestMethods;
+use args::Args;
 
-const DEFAULT_IP: &str = "127.0.0.1";
+const IP_ADDRESS: &str = "127.0.0.1";
```
Comment (nelup20) [inline src/main.rs]: @vuong-buihv Only realized after the previous merge/PR, but won't IP always be localhost/127.0.0.1?
Therefore the user can't change via args (but PORT should definitely be modifiable 👍).

Comment (vuong-buihv) [reply src/main.rs]: IP can be modified and be something else than 127.0.0.1

File: src/args.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+use clap::Parser;
+
+#[derive(Parser)]
+#[clap(version)]
+pub(crate) struct Args {
+
```
Comment (vuong-buihv) [inline src/args.rs:6]: The comments /// should be added as they are shown when we do --help or -h

File: src/request_methods.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    pub(crate) fn get_request_type(request: &str) -> &str {
+        if request.starts_with(RequestMethods::GET){
+            return RequestMethods::GET;
+        }
+
+        if request.starts_with(RequestMethods::POST){
+            return RequestMethods::POST;
+        }
+
+        return RequestMethods::UNKNOWN;
```
Comment (vuong-buihv) [inline src/request_methods.rs]: If it's on the last line then another way to return something in Rust is to just type the thing to return without return and the semicolon

**Review by vuong-buihv — CHANGES_REQUESTED** [review]

**Review by vuong-buihv — APPROVED** [review]

**Pull request merged**